watertap/costing/unit_models/electrocoagulation.py

```diff
@@ -0,0 +1,286 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pyomo.environ as pyo
+from ..util import (
+    register_costing_parameter_block,
+    make_capital_cost_var,
+    make_fixed_operating_cost_var,
+)
+
+# Costing equations from:
+
+# Saad et al., 2022 - doi.org/10.1016/j.jksus.2022.102015;
+# Smith, 2005 - doi.org/10.1205/cherd.br.0509
+# Reference for Al cost: Anuf et al., 2022 - https://doi.org/https://doi.org/10.1016/j.jwpe.2022.103074
+
+
+def build_aluminum_cost_param_block(blk):
+    costing = blk.parent_block()
+    blk.cost = pyo.Param(
+        initialize=2.23,
+        units=pyo.units.USD_2021 / pyo.units.kg,  # Anuf et al., 2022
+        mutable=True,
+        doc="Cost of aluminum plate per kg",
+    )
+    costing.register_flow_type("aluminum", blk.cost)
+
+
+def build_iron_cost_param_block(blk):
+    costing = blk.parent_block()
+    blk.cost = pyo.Param(
+        initialize=3.41,
+        units=pyo.units.USD_2021 / pyo.units.kg,  # Anuf et al., 2022
+        mutable=True,
+        doc="Cost of iron plate per kg",
+    )
+    costing.register_flow_type("iron", blk.cost)
```

Comment:
I wonder if there should be a custom electrode material in case the user wants to use a different material and not have to settle for the misnomer of aluminum or iron. How about something like `custom_electrode`?

Reply:
It is a good idea and is certainly implementable, but a user could do this with the existing framework by just changing the proper parameter values manually at the flowsheet level.

Actually, that would be essentially how I would want to implement it. I'm not sure how else you could implement a customizable config option without requiring the user to change the materials-specific parameters anyway.